src/Language/Fixpoint/Parse.hs

```diff
@@ -579,18 +960,19 @@ sortFunP
   <|> (fTyconSort <$> fTyConP)
 -}
 
+-- | Parser for sorts, parameterised over the parser for arguments.
+--
+-- TODO, Andres: document the parameter better.
+--
 sortP' :: Parser [Sort] -> Parser Sort
 sortP' appArgsP
-   =  parens sortP
-  <|> (reserved "func" >> funcSortP)
-  <|> (fAppTC listFTyCon . single <$> brackets sortP)
-  <|> bvSortP
+   =  parens sortP -- parenthesised sort, starts with "("
+  <|> (reserved "func" >> funcSortP) -- function sort, starts with "func"
+  <|> (fAppTC listFTyCon . pure <$> brackets sortP)
+  <|> bvSortP -- Andres: this looks unreachable, as it starts with "("
```

Comment:
hmm that's strange, but we do have some tests for it, search for `(BitVec Size32)` in tests/ so if they pass it should be ok?

Reply:
I think `bvSortP` is indeed unreachable. But if you look at how `bvSortP` is defined, then it is:
```
bvSortP :: Parser Sort
bvSortP = mkSort <$> (bvSizeP "Size32" S32 <|> bvSizeP "Size64" S64)
  where
    bvSizeP ss s = do
      parens (reserved "BitVec" >> reserved ss)
      return s
```
So it recognises `(BitVec Sort32)` and `(BitVec Sort64)`. The strings also parse via the `(fAppTC <$> fTyConP <*> appArgsP)` option of this parser. So I think this is just redundant.

Reply:
Actually, I think I'll comment out this branch, as I'm rather confident it is not having any effect anyway, and tests pass withou it.